mautic/mautic
# Speed improvement with Is read date sent index

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Performance
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This query took over 16 minutes on one of the Mautic instances we maintain that has email_stats with 27M rows:

```sql
SELECT COUNT(t.id) AS count FROM mautic_email_stats t WHERE (t.is_read = '1') AND (t.date_sent BETWEEN '2018-10-17 00:00:00' AND '2018-11-16 11:53:18');
```

It's used to generate some pie charts on dashboard and reports. Both of the columns in the query has its index, but MySql use only 1 of them.

This PR creates combined index which speeds up the query from 16 minutes to 400 ms.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Hard to do without very big email_stats table.

#### Steps to test this PR:
1. Ensure that you have the email stats pie chart on the dashboard.
2. Run `app/console doctrine:migrations:migrate`
2. Dashboard should still load correctly.
